src/codegen/codegen_c_visitor.cpp

```diff
@@ -3315,9 +3323,20 @@ void CodegenCVisitor::print_nrn_init(bool skip_init_check) {
 }
 
 
+void CodegenCVisitor::print_nrn_constructor() {
+    printer->add_newline(2);
+    CodegenCVisitor::print_global_function_common_code(BlockType::Constructor);
+    if (info.constructor_node != nullptr) {
+        const auto& block = info.constructor_node->get_statement_block();
+        print_statement_block(*block.get(), false, false);
+    }
+    printer->end_block(1);
+}
+
+
 void CodegenCVisitor::print_nrn_destructor() {
     printer->add_newline(2);
-    print_global_function_common_code(BlockType::Destructor);
+    CodegenCVisitor::print_global_function_common_code(BlockType::Destructor);
```

Comment:
Maybe something like
```
diff --git a/src/codegen/codegen_ispc_visitor.cpp b/src/codegen/codegen_ispc_visitor.cpp
index b2822f10..7646001f 100644
--- a/src/codegen/codegen_ispc_visitor.cpp
+++ b/src/codegen/codegen_ispc_visitor.cpp
@@ -357,6 +357,9 @@ void CodegenIspcVisitor::print_procedure(const ast::ProcedureBlock& node) {
 
 
 void CodegenIspcVisitor::print_global_function_common_code(BlockType type) {
+    if (wrapper_codegen) {
+        return CodegenCVisitor::print_global_function_common_code(type);
+    }
     std::string method = compute_method_name(type);
 
     auto params = get_global_function_parms(ptr_type_qualifier());
```
makes more sense? Otherwise if we need to call in another point `print_global_function_common_code` again we would have to remember to call the one from `CodegenCVisitor`

Reply:
ok

Reply:
done